src/components/menus/bus-menu.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+    ListItemIcon,
+    ListItemText,
+    Menu,
+    MenuItem,
+    Typography,
+} from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import BoltIcon from '@mui/icons-material/Bolt';
+import { FormattedMessage } from 'react-intl';
+import { FunctionComponent, useCallback, useMemo } from 'react';
+import {
+    isNodeBuilt,
+    isNodeReadOnly,
+} from 'components/graph/util/model-functions';
+import { useSelector } from 'react-redux';
+import { ReduxState } from 'redux/reducer.type';
+import { useIsAnyNodeBuilding } from 'components/utils/is-any-node-building-hook';
+import { ComputingType } from 'components/computing-status/computing-type';
+import { RunningStatus } from 'components/utils/running-status';
+
+interface BusMenuProps {
+    busId: string;
+    handleRunShortcircuitAnalysis: (busId: string) => void;
+    position: [number, number];
+    closeBusMenu: () => void;
+}
+
+const useStyles = makeStyles((theme) => ({
+    menu: {
+        minWidth: 300,
+        maxHeight: 800,
+        overflowY: 'visible',
+    },
+    menuItem: {
+        // NestedMenu item manages only label prop of string type
+        // It fix paddings itself then we must force this padding
+        // to justify menu items texts
+        paddingLeft: '12px',
+    },
+}));
+
+export const BusMenu: FunctionComponent<BusMenuProps> = ({
```

Comment:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/

"Why is React.FC discouraged? What about React.FunctionComponent or React.VoidFunctionComponent?
However, the general consensus today is that React.FunctionComponent (or the shorthand React.FC) is [discouraged](https://github.com/facebook/create-react-app/pull/8177). This is a nuanced opinion of course"

do we have an opinion on this ?

Reply:
After debating, we chose to use FunctionComponent